system/include/emscripten/bind.h

```diff
@@ -1390,6 +1422,50 @@ namespace emscripten {
             ;
     }
 
+    template<typename T,typename Invalid=T,typename VecType=std::vector<T>>
+    class_<VecType> register_vector_and_iterator(const char* name) {
+        std::string sname=name;
+
+        value_object<internal::IteratorItem<VecType>>((sname + "IteratorItem").c_str())
+            .field("done", &internal::IteratorItem<VecType>::done)
+            .field("value", &internal::IteratorItem<VecType>::value)
+        ;
+
+        class_<internal::ContainerIterator<VecType,Invalid>>((sname + "Iterator").c_str())
+            .function("next", &internal::ContainerIterator<VecType,Invalid>::next)
+            ;
+
+        return register_vector<T,VecType>(name)
```

Comment:
Can you try to merge register_vector_and_iterator and register_list_and_iterator into one function `register_iterator`?  There is a bunch of duplication between the two functions.